products/privateca/api.yaml

```diff
@@ -0,0 +1,265 @@
+
+--- !ruby/object:Api::Product
+name: Privateca
+display_name: Certificate Authority
+versions:
+  - !ruby/object:Api::Product::Version
+    name: beta
+    base_url: https://privateca.googleapis.com/v1beta1/
+scopes:
+  - https://www.googleapis.com/auth/cloud-platform
+apis_required:
+  - !ruby/object:Api::Product::ApiReference
+    name: Certificate Authority API
+    url: https://console.cloud.google.com/apis/api/privateca.googleapis.com
+async: !ruby/object:Api::OpAsync
+    operation: !ruby/object:Api::OpAsync::Operation
+      base_url: '{{op_id}}'
+      path: 'name'
+      wait_ms: 1000
+    result: !ruby/object:Api::OpAsync::Result
+      path: 'response'
+      resource_inside_response: true
+    status: !ruby/object:Api::OpAsync::Status
+      path: 'done'
+      complete: True
+      allowed:
+        - True
+        - False
+    error: !ruby/object:Api::OpAsync::Error
+      path: 'error'
+      message: 'message'
+objects:
+  # CertificateAuthority
+  - !ruby/object:Api::Resource
+    name: 'CertificateAuthority'
+    description: |
+        A CertificateAuthority represents an individual Certificate Authority. A
+        CertificateAuthority can be used to create Certificates.
+    base_url: projects/{{project}}/locations/{{location}}/certificateAuthorities
+    create_url: projects/{{project}}/locations/{{location}}/certificateAuthorities?certificateAuthorityId={{certificate_authority_id}}
+    self_link: projects/{{project}}/locations/{{location}}/certificateAuthorities/{{certificate_authority_id}}
+    min_version: beta
+    create_verb: :POST
+    delete_url: '{{name}}:scheduleDelete'
+    delete_verb: :POST
+    references: !ruby/object:Api::Resource::ReferenceLinks
+      guides:
+        'Official Documentation':
+          'https://cloud.google.com/certificate-authority-service'
+      api: 'https://https://cloud.google.com/certificate-authority-service/docs/reference/rest'
+    properties:
+      - !ruby/object:Api::Type::String
+        name: location
+        description: Location of the Certificate Authority.
+        required: true
+        url_param_only: true
+      - !ruby/object:Api::Type::String
+        name: certificateAuthorityId
+        description: GCP region of the Realm.
+        input: true
+        required: true
+        url_param_only: true
+      - !ruby/object:Api::Type::String
+        name: 'name'
+        description: |
+            The resource name for this CertificateAuthority in the format
+            projects/*/locations/*/certificateAuthorities/*.
+        output: true
+      - !ruby/object:Api::Type::Enum
+        name: 'type'
+        description: The Type of this CertificateAuthority.
+        input: true
+        values:
+          - :SELF_SIGNED
+        default_value: :SELF_SIGNED
+      - !ruby/object:Api::Type::Enum
+        name: 'tier'
+        description: The Tier of this CertificateAuthority.
+        input: true
+        values:
+          - :ENTERPRISE
+          - :DEVOPS
```

Comment:
`TIER_UNSPECIFIED`  ?

Reply:
TIER_UNSPECIFIED would cause the creation to fail (because it must be explicit); I thought maybe it made sense to enforce this choice with Terraform, but I'm not sure what the common practices is. Let me know if the full enum is preferable.

Reply:
this makes sense